src/system/ExtensionsModule/Api/VariableApi.php

```diff
@@ -0,0 +1,265 @@
+    }
+
+    /**
+     * Get an extension variable.
+     * @api Core-2.0
+     *
+     * @param string $extensionName The name of the extension or pseudo-extension (e.g., 'ZikulaUsersModule', 'ZConfig', '/EventHandlers').
+     * @param string $variableName The name of the variable.
+     * @param mixed $default The value to return if the requested var is not set.
+     *
+     * @return mixed - extension variable value
+     */
+    public function get($extensionName, $variableName, $default = false)
+    {
+        if (empty($extensionName) || !is_string($extensionName) || empty($variableName) || !is_string($variableName)) {
+            throw new \InvalidArgumentException();
+        }
+        if (!$this->isInitialized) {
+            $this->initialize();
+        }
+
+        if (isset($this->variables[$extensionName]) && array_key_exists($variableName, $this->variables[$extensionName])) {
+            return $this->variables[$extensionName][$variableName];
+        }
+
+        return $default;
+    }
+
+    /**
+     * Get all the variables for an extension.
+     * @api Core-2.0
+     *
+     * @param $extensionName
+     * @return array
+     */
+    public function getAll($extensionName)
+    {
+        if (empty($extensionName) || !is_string($extensionName)) {
+            throw new \InvalidArgumentException();
+        }
+        if (!$this->isInitialized) {
+            $this->initialize();
+        }
+
+        return isset($this->variables[$extensionName]) ? $this->variables[$extensionName] : [];
+    }
+
+    /**
+     * Set an extension variable.
+     * @api Core-2.0
+     *
+     * @param string $extensionName The name of the extension.
+     * @param string $variableName The name of the variable.
+     * @param string $value The value of the variable.
+     *
+     * @return boolean True if successful, false otherwise.
+     */
+    public function set($extensionName, $variableName, $value = '')
+    {
+        if (empty($extensionName) || !is_string($extensionName) || empty($variableName) || !is_string($variableName)) {
+            throw new \InvalidArgumentException();
+        }
+        if (!$this->isInitialized) {
+            $this->initialize();
+        }
+
+        $entities = $this->repository->findBy(['modname' => $extensionName, 'name' => $variableName]);
+        if (count($entities) > 1 || count($entities) == 0) {
+            foreach ($entities as $entity) {
+                // possible duplicates exist. remove all (refs #2385)
+                $this->repository->remove($entity);
+            }
+            $entity = new ExtensionVarEntity();
+            $entity->setModname($extensionName);
+            $entity->setName($variableName);
+        } else {
+            $entity = $entities[0];
+        }
+        $entity->setValue($value);
+        $this->repository->persistAndFlush($entity);
+        $this->variables[$extensionName][$variableName] = $value;
+
+        return true;
```

Comment:
One might argue that nothing should be returned, because:
- it returns true all the time anyways
- if a developer cleanly uses the set function, they are forced to check for the return type and if it's false do... what?